utils/GRI_bed_filter/version1_gene-only/gri_v1.R

```diff
@@ -0,0 +1,248 @@
+# BED region =  protein_coding genes
+#             + HGMD mutations (50 flanking bp)
+#             + Cinvar mutations (50 flanking bp)
+#
+# gencode data:
+# https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_46/gencode.v46.basic.annotation.gtf.gz
+# https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_46/GRCh37_mapping/gencode.v46lift37.basic.annotation.gtf.gz
+#
+# HGMD data:
+# hgmd2022 from Hwan
+#
+# Clinvar data:
+# https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/
+# https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh37/
+
+library(VariantAnnotation)
+library(BSgenome.Hsapiens.UCSC.hg38)
+library(BSgenome.Hsapiens.UCSC.hg19)
+library(httr)
+library(dplyr)
+library(data.table)
+
+genome_version = "hg38"
+# genome_version="hg19"
+
+
+# data preparation --------------------------------------------------------
+
+## hgmd local file name
+hgmd_hg38 = "HGMD_Pro_2022.2_hg38.vcf.gz"
+hgmd_hg19 = "HGMD_Pro_2022.2_hg19.vcf.gz"
+
+## use different data based on genome version
+if (genome_version == "hg38") {
```

Comment:
I think the cod for both hg38 and hg19 are almost the same can you make it into one function then call twice for both versions?

Reply:
Sure